src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorCodeDocumentExtensions.cs

```diff
@@ -34,6 +38,20 @@ public static SourceText GetCSharpSourceText(this RazorCodeDocument document)
         return sourceText.AssumeNotNull();
     }
 
+    public static SyntaxTree GetCSharpSyntaxTree(this RazorCodeDocument document, CancellationToken cancellationToken = default)
```

Comment:
I think it would be good to add a doc comment to this mentioning it parses C# directly, and therefore shouldn't be called in cohosting. Alternatively, perhaps renaming it to `ParseCSharpToSyntaxTree` or something would make it apparent just by its name.

Also looks like this is the only CancellationToken parameter you made optional. Was that intentional?

Reply:
Great suggestions! Thanks!

Reply:
I decided to move this to be a static method on `DocumentSnapshot`. That makes it clearer that it should only be called outside of the co-hosting universe. `TestDocumentSnapshot` can still easily call it to share the implementation.

Reply:
Sounds perfect